packages/sui-pde/src/adapters/optimizely/index.js

```diff
@@ -0,0 +1,85 @@
+import optimizelySDK from '@optimizely/optimizely-sdk'
+import integrations from './integrations'
+
+const DEFAULT_OPTIONS = {
+  autoUpdate: true,
+  updateInterval: 60 * 1000, // 60 seconds
+  logLevel: 'info'
+}
+
+const DEFAULT_TIMEOUT = 500
+
+export default class OptimizelyAdapter {
+  /**
+   * @constructor
+   * @param {object} param
+   * @param {object} param.optimizely OptimizelyInstance returned by createOptimizelyInstance method
+   * @param {string=} param.userId
+   * @param {object} param.activeIntegrations segment activated by default
+   */
+  constructor({optimizely, userId, activeIntegrations = {segment: true}}) {
+    if (!optimizely) {
+      throw new Error(
+        'optimizely instance is mandatory to use OptimizelyAdapter'
+      )
+    }
+
+    this._optimizely = optimizely
+    this._userId = userId?.toString()
+
+    if (this._isActivated()) {
+      integrations.forEach(integration =>
+        integration({
+          activeIntegrations,
+          optimizelyInstance: optimizely
+        })
+      )
+    }
+  }
+
+  /**
+   * @param {object} param
+   * @param {string} param.sdkKey
+   * @param {object} [param.options] https://docs.developers.optimizely.com/full-stack/docs/initialize-sdk-javascript-node
+   */
+  static createOptimizelyInstance({options: optionParameter, sdkKey}) {
+    const options = {...DEFAULT_OPTIONS, ...optionParameter}
+    optimizelySDK.setLogLevel(options.logLevel)
+    optimizelySDK.setLogger(optimizelySDK.logging.createLogger())
+    const optimizelyInstance = optimizelySDK.createInstance({
+      sdkKey,
+      datafileOptions: options
+    })
+
+    return optimizelyInstance
+  }
+
+  _isActivated() {
```

Comment:
I think it could be clear if function name specify that it is related to user existance.
Method `activateExperiment` on L:75 sounds weird to me. I mean, If I don't look in depth, `activateExperiment` function checks if `_isActivated` and if it is not, it returns 'default` whenever it should activate test